pkg/cli/admin/inspect/resource.go

```diff
@@ -94,30 +95,114 @@ func InspectResource(info *resource.Info, context *resourceContext, o *InspectOp
 		}
 		return nil
 
-	case schema.GroupResource{Group: "route.openshift.io", Resource: "routes"}:
+	case routev1.GroupVersion.WithResource("routes").GroupResource():
 		if err := inspectRouteInfo(info, o); err != nil {
 			return err
 		}
 		return nil
+
+	case admissionregistrationv1.SchemeGroupVersion.WithResource("mutatingwebhookconfigurations").GroupResource():
+		if err := gatherMutatingAdmissionWebhook(context, info, o); err != nil {
+			return err
+		}
+		return nil
+
+	case admissionregistrationv1.SchemeGroupVersion.WithResource("validatingwebhookconfigurations").GroupResource():
+		if err := gatherValidatingAdmissionWebhook(context, info, o); err != nil {
+			return err
+		}
+		return nil
+
+	case apiextensionsv1.SchemeGroupVersion.WithResource("customresourcedefinitions").GroupResource():
+		if err := gatherCustomResourceDefinition(context, info, o); err != nil {
+			return err
+		}
+		return nil
+
 	default:
-		unstr, ok := info.Object.(*unstructured.Unstructured)
-		if ok {
-			// obtain associated objects for the current resource
-			if err := gatherRelatedObjects(context, unstr, o); err != nil {
-				return err
+		return gatherGenericObject(context, info, o)
+	}
+}
+
+type listAccessor interface {
+	addItem(obj interface{}) error
+}
+
+func newListAccessor(structuredList interface{}) (listAccessor, error) {
+	switch castObj := structuredList.(type) {
+	case *admissionregistrationv1.MutatingWebhookConfigurationList:
+		return &mutatingWebhookConfigList{castObj}, nil
+
+	case *admissionregistrationv1.ValidatingWebhookConfigurationList:
+		return &validatingWebhookConfigList{castObj}, nil
+
+	case *corev1.SecretList:
+		return &secretList{castObj}, nil
+
+	case *routev1.RouteList:
+		return &routeList{castObj}, nil
+
+	default:
+		return nil, fmt.Errorf("unhandledStructuredListType: %T", castObj)
+
+	}
+}
+
+func toStructuredObject[T any, TList any](obj runtime.Object) (runtime.Object, error) {
```

Comment:
my first experiment with generics.  Any better ideas for doing this?

Reply:
Theoretically this will be ok-ish, but `T` and `TList` do NOT have to be correlated, iow. I can pass `admissionregistrationv1.MutatingWebhookConfiguration` twice here and the compiler will happily accept that, which is what you're trying to prevent from happening, right?